service-account/aws/main.tf

```diff
@@ -0,0 +1,99 @@
+
+
+locals {
+  prefix   = var.prefix
+  tags     = merge(var.tags, { module = "amazon-sqs" })
+  oidc_arn = var.oidc_provider_arn
+  oidc_url = trimprefix(var.oidc_issuer_url, "https://")
+}
+
+resource "aws_iam_role" "armonik" {
+  name = "${local.prefix}-eks-pod-identity-armonik"
+  assume_role_policy = jsonencode({
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Effect = "Allow"
+        Principal = {
+          Federated = local.oidc_arn
+        }
+        Action = "sts:AssumeRoleWithWebIdentity"
+        Condition = {
+          StringEquals = {
+            "${local.oidc_url}:aud" = "sts.amazonaws.com"
+            "${local.oidc_url}:sub" = [
+              "system:serviceaccount:${var.namespace}:${var.service_account_name}",
+            ]
+          }
+        }
+      }
+    ]
+  })
+  tags = local.tags
+}
+
+resource "aws_iam_policy_attachment" "armonik_decrypt_object" {
+  name       = "${local.prefix}-s3-encrypt-decrypt-armonik"
+  roles      = [aws_iam_role.armonik.name]
+  policy_arn = var.decrypt_policy_arn
+}
+
+data "aws_iam_policy_document" "sqs" {
+  statement {
+    sid = "SqsAdmin"
+    actions = [
+      "sqs:*"
+    ]
+    effect    = "Allow"
+    resources = ["*"]
+  }
+}
+
+resource "aws_iam_policy" "sqs" {
+  name_prefix = "${local.prefix}-sqs-admin"
+  description = "Policy for allowing SQS admin access"
+  policy      = data.aws_iam_policy_document.sqs.json
+  tags        = local.tags
+}
+
+resource "aws_iam_policy_attachment" "sqs" {
+  name       = "${local.prefix}-sqs"
+  roles      = [aws_iam_role.armonik.name]
+  policy_arn = aws_iam_policy.sqs.arn
+}
+
+data "aws_iam_policy_document" "s3" {
+  statement {
+    sid = "S3Admin"
+    actions = [
+      "s3:*"
+    ]
+    effect    = "Allow"
+    resources = ["*"]
+  }
+}
+
+resource "aws_iam_policy" "s3" {
+  name_prefix = "${local.prefix}-s3-admin"
+  description = "Policy for allowing S3 admin access"
+  policy      = data.aws_iam_policy_document.s3.json
+  tags        = local.tags
+}
+
+resource "aws_iam_policy_attachment" "s3" {
+  name       = "${local.prefix}-s3"
+  roles      = [aws_iam_role.armonik.name]
+  policy_arn = aws_iam_policy.s3.arn
+}
```

Comment:
Is it linked to SQS introduction ?

Reply:
It's not for SQS, I wasn't sure if it was needed for some new part of ArmoniK.Core so I left it there. I removed it now and if it turns out that that's the case I can just add it in another PR.

Reply:
Adding it back, and going to change the PR to be about adding a service account for AWS. I'm putting it on draft until I deploy and test the specific permissions I added in.